design/ceph/object/store.md

```diff
@@ -712,3 +712,95 @@ For external CephCephObjectStores (i.e., when `spec.gateway.externalRgwEndpoints
+This tradeoff favors reliability over flexibility.
+
+Examples:
+    - Rook sets configs like `rgw_zone` which are critical for a functional CephObjectStore. Users
+    would not be able to override this value.
+    - Rook sets `rgw_dns_name` via the `hosting` config section. User inputs to `rgw_dns_name`
+    wouldn't take affect when `hosting` configs are set.
+
+If possible, Rook should make log when it overrides a user's `cephConfig` based on another config or
+internal decision. This may not be easily feasible when Rook applies CLI flags to the RGW
+deployment.
+
+Rook will apply each configuration under the client scope of the shared RGW client.
+i.e., `client.rook-ceph-rgw-<store-name>`. This will also help ensure that configs changed don't
+affect other Rook-managed Ceph daemons, critically mons, osds, and mds.
+
+Configs set here will be equivalent to a Ceph config section like the following ini.
+
+```ini
+[client.rook-ceph-rgw-<store-name>]
+rgw_enable_apis = s3, s3website, swift, swift_auth, sts, iam, notifications
+rgw_gc_max_concurrent_io = "4"
+debug_rgw = "20"
+```
+
+Rook will not perform validation to ensure configs set in the CephObjectStore `cephConfig` section
+do not conflict with CephCluster `config` options. Users should refrain from setting
+`client.rook-ceph-rgw-*` configs in the CephCluster when this feature is used. Documentation should
+note this for users. Rook can revisit this decision in the future to add cross-CRD validation if it
+becomes a strong user desire/need.
+
+In order to prevent misuse or abuse of this feature, Rook will perform some high-level validation of
+configurations. For example, users will be prevented from setting the
+[Ceph general config](https://docs.ceph.com/en/reef/rados/configuration/general-config-ref/)
+`admin_socket` which might affect RGW's ability to run correctly in the container. Rook will
+disallow all configurations by default, and use an allow-list architecture to allow user configs
+that are reasonably in scope for object stores.
+
+Critically, users should be able to modify RGW configurations. Ceph prefixes all RGW-related
+configurations with `rgw_`, so Rook will allow-list all `rgw_`-prefixed configs.
+
+Users are often asked to set Ceph log levels for debugging challenging issues. Ceph has an `rgw`
+[debug subsystem](https://docs.ceph.com/en/reef/rados/troubleshooting/log-and-debug/). Rook can
+allow users to set RGW log levels per-objectstore by allowing `debug_rgw` as a prefix also. The
+prefix will allow `debug_rgw_access`, and similar options.
+
+These validations are simple to implement and maintain. If Rook finds that users need configurations
+outside of these allowed bounds, maintainers can decide to add broad or specific allowances then, or
+open up all configs to this mechanism at future discretion. Implementing restrictions in the future
+would necessitate breaking change notices, so it is preferred to implement restrictions early.
```

Comment:
@jhoblitt and @travisn what are your thoughts on the latest update for input validation. I do tend to agree with Josh that bounding the feature set now will ensure users aren't misusing the system, hopefully preventing us from making any breaking changes for them in the future.

The validations I'm proposing are simple to implement/test/maintain: just allow-list `rgw_*` or `debug rgw*` keys, and disallow the rest. Given the simplicity and benefit of providing **some** guardrails, I think it's worth it.

Reply:
As I thought about this some more, I think having the subject of the configs be limited to the `client.rook-ceph-rgw-*` user will be enough to ensure the configs aren't going to break the rest of the Ceph cluster, and I got exhausted from trying to think of all the what-ifs. Ultimately, there is risk we need to make changes with either choice now, but since this is aimed at user flexibility, we may as well go all-in.